### ISSUE TITLE
Update request: haskellPackages.xmonad-contrib 0.18.0 → 0.18.1

### ISSUE THREAD

**Issue by u2x1** [NixOS/nixpkgs#340712]
- Package name: `haskellPackages.xmonad-contrib`
- Latest released version: `0.18.1`

- Current version on the unstable channel: `0.18.0`
- Current version on the stable/release channel: `0.18.0`

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**

@dschrempf @ivanbrennan @peti @slotThe 

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by slotThe** [comment]
AFAIK it's already in https://github.com/NixOS/nixpkgs/pull/339272

**Comment by sternenseemann** [comment]
`xmonad-contrib` has a hard dependency on xmonad `>= 0.18`, so we won't be able to ship it as the default version since we inherit 0.17 from Stackage LTS. I expect you'll be able to use xmonad-contrib `>= 0.18.1` by using `haskellPackages.xmonad_0_18_0`, though.

**Comment by u2x1** [comment]
> I expect you'll be able to use xmonad-contrib `>= 0.18.1` by using `haskellPackages.xmonad_0_18_0`, though.


Thank you.

I am still fairly new to nix. Can you please tell me what should I do next?

```
windowManager.xmonad = {
  enable = true;
  enableContribAndExtras = true;
  extraPackages = haskellPackages : [
    haskellPackages.xmonad-contrib
    haskellPackages.xmonad-extras
    haskellPackages.xmobar
    haskellPackages.xmonad_0_18_0
  ];
};
```

Is that enough?

**Comment by dschrempf** [comment]
Hm, I think this will not change the underlying `xmonad` build and provided set of libraries. As far I can see, you need to use a Haskell package set with `xmonad` version 0.18. For example,

    services.xserver.windowManager.xmonad.haskellPackages = pkgs.haskell.packages.ghc98

Then you have `xmonad 0.18`. But then you still need to use `xmonad-contrib` v 0.18.1. I am not sure if this is available.

**Comment by u2x1** [comment]
> Hm, I think this will not change the underlying `xmonad` build and provided set of libraries. As far I can see, you need to use a Haskell package set with `xmonad` version 0.18.

Oh, thanks for the information!


> But then you still need to use `xmonad-contrib` v 0.18.1. I am not sure if this is available.

No, I just tried and sure it isn't available. But can I have an overlay or something to patch it myself?

**Comment by sternenseemann** [comment]
It's going to be available after #339272 (see also https://github.com/NixOS/nixpkgs/pull/339272/commits/9d2a706ae7b358268e77a281a86facbd271df515).

**Comment by sushi-shi** [comment]
@u2x1 this did the trick for me:
```
       services.xserver.windowManager.xmonad.haskellPackages = pkgs.haskell.packages.ghc98.override {
         overrides = self: super: {
           xmonad-contrib = self.callHackageDirect {
              pkg = "xmonad-contrib";
              ver = "0.18.1";
              sha256 = "sha256-3N85ThXu3dhjWNAKNXtJ8rV04n6R+/rGeq5C7fMOefY=";
           } {};
         };
       };
```
